main.sh

```diff
@@ -342,9 +344,16 @@ if [[ "${INPUT_TAR/all/${platform}}" == "${platform}" ]] || [[ "${INPUT_ZIP/all/
     cwd=$(pwd)
     tmpdir=$(mktemp -d)
     mkdir "${tmpdir:?}/${archive}"
-    filenames=("${bins[@]}")
+    filenames=()
     for bin_exe in "${bins[@]}"; do
-        cp "${target_dir}/${bin_exe}" "${tmpdir}/${archive}"/
+        if [[ -n "${bin_leading_dir}" ]]; then
+            x mkdir -p "${tmpdir}/${archive}/${bin_leading_dir}"/
+            x cp "${target_dir}/${bin_exe}" "${tmpdir}/${archive}/${bin_leading_dir}"/
+            filenames+=("${bin_leading_dir%%/*}")
```

Comment:
This adds `"${bin_leading_dir%%/*}"` many times when there are multiple binaries, right? I suspect that has not been tested to work correctly.

Perhaps it would be better to add the directory before the loop as before.

Reply:
you are right, fixing.

Reply:
Update: move `filenames` out of the for loop, set the `filenames` before this for loop.

Reply:
Manual testing the generated packages.

Reply:
Update:

After fixed the `filenames` and `mkdir` issues, I manually tested this branch again in my rust/cargo [example CI project](https://github.com/linrongbin16/cargo-ci-template.rs).

In [latest commits](https://github.com/linrongbin16/cargo-ci-template.rs/blob/53b5976b3453eb0462cb50caa25083bc442c9d4a/.github/workflows/upload-rust-binary-build.yml#L25-L79), I add multiple bins to test, and the release is: https://github.com/linrongbin16/cargo-ci-template.rs/releases/tag/v0.4.50. They're all good now.

> I actually waste more 2 hours to finish this manual testing. Because I found the `leading-dir` is always not working in my PR's branch. Finally I found that, the root cause is my macbook, the `tar` on macbook will **eat** the leading directory. So all good now.